table.cc

```diff
@@ -1287,6 +1288,8 @@ table::compact_sstables(sstables::compaction_descriptor descriptor, bool cleanup
             if (release_exhausted) {
                 release_exhausted(old_ssts);
             }
+            // This is done in the background, so we can consider this compaction completed.
+            (void)this->delete_compacted_sstables(old_ssts);
         };
```

Comment:
Maybe we shouldn't do it in the background, and consider all compactions complete after they deleted their inputs. If deletion is slow for some reason, and we don't wait for it, we can run out of space. Also background jobs are like land mines except they move around and actively seek you.

Reply:
I like that direction. It would also help with implementing a disk-space controller.